shell/core/plugin.ts

```diff
@@ -161,21 +162,47 @@ export class Plugin implements IPlugin {
    * Adds a tab to the UI
    */
   addTab(where: string, when: LocationConfig | string, tab: Tab): void {
-    this._addUIConfig(ExtensionPoint.TAB, where, when, tab);
+    this._addUIConfig(ExtensionPoint.TAB, where, when, this._createAsyncComponent(tab));
   }
 
   /**
    * Adds a panel/component to the UI
    */
   addPanel(where: string, when: LocationConfig | string, panel: Panel): void {
-    this._addUIConfig(ExtensionPoint.PANEL, where, when, panel);
+    this._addUIConfig(ExtensionPoint.PANEL, where, when, this._createAsyncComponent(panel));
   }
 
   /**
    * Adds a card to the to the UI
    */
   addCard( where: string, when: LocationConfig | string, card: Card): void {
-    this._addUIConfig(ExtensionPoint.CARD, where, when, card);
+    this._addUIConfig(ExtensionPoint.CARD, where, when, this._createAsyncComponent(card));
+  }
+
+  /**
+   * Wraps a component from an extensionConfig with defineAsyncComponent and
+   * markRaw. This prepares the component to be loaded dynamically and prevents
+   * Vue from making the component reactive.
+   *
+   * @param extensionConfig The extension configuration containing a component
+   * to render.
+   * @returns A new object with the same properties as the extension
+   * configuration, but with the component property wrapped in
+   * defineAsyncComponent and markRaw. If the extension configuration doesn't
+   * have a component property, it returns the extension configuration
+   * unchanged.
+   */
+  private _createAsyncComponent(extensionConfig: Card | Panel | Tab) {
+    const { component } = extensionConfig;
+
+    if (!component) {
+      return extensionConfig;
+    }
+
+    return {
+      ...extensionConfig,
+      component: markRaw(defineAsyncComponent(component as () => Promise<Component>)),
```

Comment:
(Moving my previous comment directly to the code as better conversation, but I forgot to copy it 🤦 )

Is there any side effect from removing reactivity with `markRaw`? If so, we should make it clear to the developers who intend to use the extensions, as they may not be familiar with Vue patterns.

Reply:
No side effect that I'm aware of. Using `markRaw` prevents Vue from treating the component as a reactive object - this can happen when a component is passed to another component via props.

Reply:
Wait, are there not plenty of these cases around?